packages/core/types/src/product/service.ts

```diff
@@ -1433,6 +1434,96 @@ export interface IProductModuleService extends IModuleService {
     sharedContext?: Context
   ): Promise<Record<string, string[]> | void>
 
+  /**
+   * This method adds a product option to a product.
+   *
+   * @param {ProductOptionProductPair} productOptionProductPair - The details of the product option and the product it should be added to.
+   * @param {Context} sharedContext - A context used to share resources, such as transaction manager, between the application and the module.
+   * @returns {Promise<{ id: string; }>} The ID of the relation between the product option and the product.
+   *
+   * @example
+   * const productOptionProductId =
+   *   await productModuleService.addProductOptionToProduct({
+   *     product_id: "prod_123",
+   *     product_option_id: "opt_123",
+   *   })
+   */
+  addProductOptionToProduct(
+    productOptionProductPair: ProductOptionProductPair,
```

Comment:
suggestion: if we can accept an array of pairs then we can make it bulk as mentioned in a previous comment